### PR TITLE
Preprocessing data split

### PR DESCRIPTION
**Pull request by Lcgunn**

### CODE REVIEW

File: preprocessing/train_val_split.ipynb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    "    writer.write_reviews(val_reviews)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO write the remaining samples to a new file, test-split.csv"
```
Comment (frejaz) [inline preprocessing/train_val_split.ipynb:58]: No todos in main

File: preprocessing.py [hunk 2]
```diff
@@ -1 +1,34 @@
 #This is for preprocessing: First will be relabeling everything and then will be splitting the data
+
+# Freya will do relabeling here:
+
+
+# Data splitting
+import torch, sklearn
```
Comment (frejaz) [inline preprocessing.py:7]: This import is not used

File: preprocessing.py [hunk 3]
```diff
@@ -1 +1,34 @@
...
+import data_reader
+import os
+import numpy as np
+
+# Open train.csv
```
Comment (frejaz) [inline preprocessing.py:12]: Task description said: "Create function which takes csv as input and splits it into train/test/val csv" - so it would be nice if this was wrapped in a function and saved the differet csvs🙏 Also, no magic numbers, so the ratio of train/validation should also be given

File: preprocessing.py [hunk 4]
```diff
@@ -1 +1,34 @@
...
+reader = data_reader.data_reader()
+csv_path = os.path.join(os.path.curdir, "dataset", "train.csv")
+reader.open_csv(csv_path)
+
+#Separate labels and features
+i = 1
+data = []
+data.append(reader.read(3000000))
```
Comment (frejaz) [inline preprocessing.py:20]: Size of data changes as we e.g. discard neutral reviews, so no hard coded values in the function

File: preprocessing.py [hunk 5]
```diff
@@ -1 +1,34 @@
...
+data.reshape((3000000,3))
+
+print(data)
+
+
+X = data[:, :-1]
+y = data[:, -1]
+print('num_samples, num_features', X.shape)
+print('labels', np.unique(y))
+print('label size', y.shape)
```
Comment (frejaz) [inline preprocessing.py:31]: save csvs somewhere - you can see how I did it in /preprocessing/reformat_data.py under my pull request if that helps, but there are other ways to do it too👌

File: preprocessing/train_val_split.ipynb [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Shuffle the input training data; use a set seed for deterministic behavior\n",
+    "reader.shuffle(random_state=9817)\n",
+    "\n",
+    "# Choose 650000 samples to be used for validation\n",
```
Comment (frejaz) [inline preprocessing/train_val_split.ipynb:39]: No magic numbers

File: preprocessing/train_val_split.ipynb [hunk 7]
```diff
@@ -0,0 +1,83 @@
+{
+ "cells": [
```
Comment (frejaz) [inline preprocessing/train_val_split.ipynb:2]: In general I'm a bit confused about the two files - what should be run to get a validation set? Is the notebook just for debugging? Or is it the other way around?

**Review by frejaz — CHANGES_REQUESTED** [review]
Some comments✨

**Comment by Lcgunn** [comment]
Superseded by #8
